src/unix/fs.c

```diff
@@ -701,8 +701,24 @@ static void uv__to_stat(struct stat* src, uv_stat_t* dst) {
   dst->st_mtim.tv_nsec = src->st_mtime_nsec;
   dst->st_ctim.tv_sec = src->st_ctime;
   dst->st_ctim.tv_nsec = src->st_ctime_nsec;
-  dst->st_birthtim.tv_sec = src->st_ctime;
-  dst->st_birthtim.tv_nsec = src->st_ctime_nsec;
+
+  /* On systems that do not support birthtime, resolve the value to the least
+   * time available (oldest of atime, mtime, ctime), disregarding nanoseconds.
+   */
+  if (src->st_atime < src->st_mtime && src->st_atime < src->st_ctime) {
```

Comment:
The actual comparison order is mostly arbitrary, but all cases are (I think, cause of `&&`) equal in comparison time at 2 checks, except for the worst case (ctime < atime < mtime). The worst case is also arbitrary, but I can change that if we think there is a better one.